src/devtools/client/debugger/src/components/Editor/useEditor.ts

```diff
@@ -453,10 +453,11 @@ function scrollToLocationHelper(
   if (selectedLocation) {
     let shouldScrollToLocation = false;
     if (
-      selectedSource != null &&
-      selectedSourceContent != null &&
-      selectedLocation?.line != null &&
-      selectedSourceContent?.value != null
+      selectedSource &&
+      selectedSourceContent &&
+      // Intentionally bail out if line is 0
+      selectedLocation?.line &&
```

Comment:
Is this really what we alway want?

Seems like there may be cases where we _want_ to scroll the editor to line 0 (e.g. if there's a comment there, or a log, etc)

Reply:
I thought lines were 1-indexed 🤔 (or at least, I'm pretty sure they are in protocol land, not sure about the editor)

Reply:
I've found and fixed some broken conditionals where we had `!value` and sometimes bailed unintentionally when `value` was 0. I think "locations" are generally _protocol_ thing, and the protocol uses 1-based line numbers and 0-based column indices for some reason 😞

Anyway, in this case– lines must have been coming in as 0 in some cases or Mark's change would not have affected anything...right?

I dunno. It's late.

Reply:
Yeah, what I saw in both pre-`useEditor` code and current code is that when I was switching tabs, `selectedLocation.line` was 0.  Not sure if that's _always_ the case, but it was at least in these cases.

Reply:
> Anyway, in this case– lines must have been coming in as 0 in some cases or Mark's change would not have affected anything...right?

I don't think that's what this sounds like:

> Yeah, what I saw in both pre-useEditor code and current code is that when I was switching tabs, selectedLocation.line was 0

When you are switching tabs you are not navigating to a protocol-provided location, probably somewhere we just hard-coded `{line: 0, column: 0}` as the default location while switching tabs, but actually that's an invalid location that should never exist.

Reply:
> When you are switching tabs you are not navigating to a protocol-provided location, probably somewhere we just hard-coded `{line: 0, column: 0}` as the default location

Ah, if so then I think we should just store `null` as the _default_ rather than an invalid/fake location. That would make more sense (and maybe make this weird check for an invalid value not relevant)

Reply:
> > Anyway, in this case– lines must have been coming in as 0 in some cases or Mark's change would not have affected anything...right?
>
> I don't think that's what this sounds like:

Not to belabor the point, but I think that's exactly what it sounded like was happening.